devito/petsc/types.py

```diff
@@ -131,15 +144,18 @@ def dtype(self):
 
     @property
     def symbolic_shape(self):
-        info = DMDALocalInfo(name='info')
-        # To access the local grid info via the DM in
-        # PETSc you use DMDAGetLocalInfo(da, &info)
-        # and then access the local no.of grid points via info.gmx, info.gmy, info.gmz
         field_from_composites = [
-            FieldFromComposite('g%sm' % d.name, info) for d in self.dimensions]
+            FieldFromComposite('g%sm' % d.name, self.dmda_info) for d in self.dimensions]
         # Reverse it since DMDA is setup backwards to Devito dimensions.
         return DimensionTuple(*field_from_composites[::-1], getters=self.dimensions)
 
+    @property
+    def dmda_info(self):
+        # To access the local grid info via the DM in
+        # PETSc you use DMDAGetLocalInfo(da, &info)
+        # and then access the local no.of grid points via info.gmx, info.gmy, info.gmz
+        return DMDALocalInfo(name='info', liveness='eager')
```

Comment:
If this will create a new `DMDALocalInfo` object each time, then `cached_property` would be better